osu.Game/Screens/Play/GameplayMenuOverlay.cs

```diff
@@ -194,14 +201,30 @@ public void OnReleased(KeyBindingReleaseEvent<GlobalAction> e)
         {
         }
 
-        private void updateRetryCount()
-        {
-            // "You've retried 1,065 times in this session"
-            // "You've retried 1 time in this session"
+        [Resolved]
+        private IGameplayClock? gameplayClock { get; set; }
+
+        [Resolved]
+        private DrawableRuleset? drawableRuleset { get; set; }
 
+        private void updateInfoText()
+        {
             playInfoText.Clear();
             playInfoText.AddText("Retry count: ");
             playInfoText.AddText(retries.ToString(), cp => cp.Font = cp.Font.With(weight: FontWeight.Bold));
+
+            if (gameplayClock != null && drawableRuleset != null)
+            {
+                double firstHitTime = drawableRuleset.Objects.FirstOrDefault()?.StartTime ?? 0;
+                //TODO: this isn't always correct (consider mania where a non-last object may last for longer than the last in the list).
```

Comment:
I'm not sure how to feel about this same TODO comment showing up in source for a third time:

https://github.com/ppy/osu/blob/6392c2007c104a37f18cdf9e22bfb295c738a38b/osu.Game/Screens/Play/HUD/SongProgress.cs#L55-L57

https://github.com/ppy/osu/blob/6392c2007c104a37f18cdf9e22bfb295c738a38b/osu.Game/Beatmaps/BeatmapUpdater.cs#L92-L94

Especially so that there are also places that are calculating this... correctly?

https://github.com/ppy/osu/blob/6392c2007c104a37f18cdf9e22bfb295c738a38b/osu.Game/Screens/Play/HUD/ArgonSongProgressGraph.cs#L29-L30

https://github.com/ppy/osu/blob/6392c2007c104a37f18cdf9e22bfb295c738a38b/osu.Game/Screens/Play/HUD/DefaultSongProgressGraph.cs#L29-L30

Kinda weird. If we want to be fast-but-maybe-wrong then maybe I'd at least split a helper or something? Not sure.

Reply:
It's almost like we should be using `BeatmapInfo.Length` in all these cases (.. and correct it while we're there to be correct), but I don't know how much coverage we have of that yet, ie. whether it has already reprocessed all old beatmaps imported to a user's library before it existed.

I'll look to combine these in a separate PR I think.